internal/provider/resource/integration_aws_parameter_store.go

```diff
@@ -0,0 +1,501 @@
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/listplanmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/planmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/stringplanmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/hashicorp/terraform-plugin-framework/types/basetypes"
+)
+
+// Ensure the implementation satisfies the expected interfaces.
+var (
+	_ resource.Resource = &IntegrationAWSParameterStoreResource{}
+)
+
+// NewProjectResource is a helper function to simplify the provider implementation.
+func NewIntegrationAwsParameterStoreResource() resource.Resource {
+	return &IntegrationAWSParameterStoreResource{}
+}
+
+// IntegrationAwsParameterStore is the resource implementation.
+type IntegrationAWSParameterStoreResource struct {
+	client *infisical.Client
+}
+
+type AwsParameterStoreMetadataStruct struct {
+	SecretAWSTag []struct {
+		Key   string `json:"key,omitempty"`
+		Value string `json:"value,omitempty"`
+	} `json:"secretAWSTag,omitempty"`
+	ShouldDisableDelete bool `json:"shouldDisableDelete,omitempty"`
+}
+
+type AwsParameterStoreOptions struct {
+	AwsTags             []awsTag `tfsdk:"aws_tags"`
+	ShouldDisableDelete bool     `tfsdk:"should_disable_delete"`
+}
+
+type awsTag struct {
+	Key   types.String `tfsdk:"key"`
+	Value types.String `tfsdk:"value"`
+}
+
+// projectResourceSourceModel describes the data source data model.
+type IntegrationAWSParameterStoreResourceModel struct {
+	AccessKeyID     types.String `tfsdk:"access_key_id"`
+	SecretAccessKey types.String `tfsdk:"secret_access_key"`
+	ProjectID       types.String `tfsdk:"project_id"`
+
+	IntegrationAuthID types.String `tfsdk:"integration_auth_id"`
+	IntegrationID     types.String `tfsdk:"integration_id"`
+	Environment       types.String `tfsdk:"environment"`
+	SecretPath        types.String `tfsdk:"secret_path"`
+	AWSPath           types.String `tfsdk:"parameter_store_path"`
+	AWSRegion         types.String `tfsdk:"aws_region"`
+
+	Options types.Object `tfsdk:"options"`
+}
+
+func extractOptionsParameterStore(ctx context.Context, diagnostics *diag.Diagnostics, inputOptions types.Object) (parsedOptions AwsParameterStoreMetadataStruct, hasError bool) {
+	metadata := AwsParameterStoreMetadataStruct{}
+
+	if !inputOptions.IsNull() && !inputOptions.IsUnknown() {
+		var options AwsParameterStoreOptions
+		diags := inputOptions.As(ctx, &options, basetypes.ObjectAsOptions{})
+		diagnostics.Append(diags...)
+		if diagnostics.HasError() {
+			return AwsParameterStoreMetadataStruct{}, true
+		}
+
+		// Load AWS tags into metadata
+		for _, tag := range options.AwsTags {
+			metadata.SecretAWSTag = append(metadata.SecretAWSTag, struct {
+				Key   string `json:"key,omitempty"`
+				Value string `json:"value,omitempty"`
+			}{
+				Key:   tag.Key.ValueString(),
+				Value: tag.Value.ValueString(),
+			})
+		}
```

Comment:
to test: check what happens with reordering of tags

Reply:
Tested with SetNestedAttribute: Nothing happens